ge/org.osate.ge.ba/src/org/osate/ge/ba/ui/properties/EmbeddedEditingControls.java

```diff
@@ -0,0 +1,84 @@
+package org.osate.ge.ba.ui.properties;
+
+import java.util.Objects;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.StyledText;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+
+class EmbeddedEditingControls {
```

Comment:
- From the name, it wasn't clear what the purpose was.
- I would prefer if it was it's own control which extends Composite. See See FilteringListSelector/NameEditor for examples but it doesn't need to be as abstract as those classes. Ideally there would be just one function that is called to refresh. Maybe not practical.
- If that is not practical:				
  - Having a version that of createStyledText() that uses it's container's layout data is very niche and doesn't make a whole lot of sense to me. Remove and just be explicit about the layout data data is desired.				
  - Add the least, make all functions not called from somewhere else private and make the ones that need to be called public. 
  - Instead of providing direct access to the styled text and button, have functions that expose the needed functionality(setting the testing ID) or add a parameter when creating.

Reply:
I made some changes, did not do a refresh function.  We can discuss if this still isn't what you had in mind.